### ISSUE TITLE
Suppressed errors aren't suppressed in QM

### ISSUE THREAD

**Issue by Daronspence** [johnbillion/query-monitor#175]
In `class-wp-http-encoding.php` in WP core, there is a purposefully suppressed error on line 58

``` php
if ( false !== ( $decompressed = @gzinflate( $compressed ) ) )
			return $decompressed;
```

On some sites where I am doing lots of HTTP requests to remote servers, this error will get triggered in Query Monitor; potentially on every page load if there is an external HTTP request to a server with that gzip configuration.

Is this something that could be addressed in QM? Should QM even show a notice for suppressed errors purposefully made in WP core? Or is there a hook I can use to suppress it on those couple of sites when I'm working on them? 

If this isn't the place to discuss this kind of thing, I apologize!

Thanks for the great plugin! :smile:

**Comment by johnbillion** [comment]
I suspect this is caused by 1f68326 which removed the `error_reporting()` check. Not sure if there's a means of detecting use of the `@` suppressor independently of this. I'll have a look into it.

**Comment by Daronspence** [comment]
You're probably right. I didn't notice this until a recent update.

Thanks for looking into it!

**Comment by crstauf** [comment]
> I suspect this is caused by 1f68326 which removed the error_reporting() check.

I just confirmed this is indeed the case. Is there a point in time in which the `error_reporting()` test can be re-instituted?

Or could QM record the system error reporting level, and then within the event handler, test if `error_reporting() > 0 AND $this->system_error_level > 0`? Something like this for `QM_Collector_PHP_Errors`:

```
public function __construct() {
	...
	$this->system_level_error = error_reporting();
	...
}

public function error_handler(...) {
	...
	if (
		error_reporting() > 0
		&& $this->system_error_level > 0
	) {

		if ( ! class_exists( 'QM_Backtrace' ) ) {
			return false;
		}

		$trace  = new QM_Backtrace( array(
			'ignore_current_filter' => false,
		) );
		$caller = $trace->get_caller();
		$key    = md5( $message . $file . $line . $caller['id'] );

		$filename = QM_Util::standard_dir( $file, '' );

		if ( isset( $this->data['errors'][$type][$key] ) ) {
			$this->data['errors'][$type][$key]->calls++;
		} else {
			$this->data['errors'][$type][$key] = (object) array(
				'errno'    => $errno,
				'type'     => $type,
				'message'  => $message,
				'file'     => $file,
				'filename' => $filename,
				'line'     => $line,
				'trace'    => $trace,
				'calls'    => 1
			);
		}
	}
	...
}
```

**Comment by johnbillion** [comment]
In [the `develop` branch](https://github.com/johnbillion/query-monitor/tree/develop), QM now collects @-suppressed errors separately. They don't trigger a highlight colour in the admin toolbar, but they do get a menu item. They're listed separately in the errors panel, too.

Take a look and see what you think.

**Comment by johnbillion** [comment]
You can easily test this, by the way, with something like the following:

```php
function my_notice() {
  $one = $two;
}

@my_notice();
```

**Comment by crstauf** [comment]
this has created a bug within the Light admin theme (#213)... will need to update stylesheet to include new `.qm-notice-suppressed` class.

**Comment by crstauf** [comment]
I really like it being separate from other PHP errors... and also that a class is added to the menu bar item, so we can activate a background color if desired.

**Comment by crstauf** [comment]
since we're not considering suppressed errors to be errors worth highlighting, perhaps `.qm-all-clear` should still be added to the top admin bar menu item?

**Comment by crstauf** [comment]
Okay, I think the only way to fix #213 would be to create a new class, `.qm-only-suppresseds`, which is added only if the "errors" in QM are limited to suppressed warnings/notices/stricts/deprecateds. I hate to be filling up the menu item and stylesheet with so many items, but this is the only solution I've been able to come up with.
